File: app/src/main/java/com/example/wellcome/utils/WellcomeDbHelper.java
package com.example.wellcome.utils;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class WellcomeDbHelper extends SQLiteOpenHelper {
    public static final int DATABASE_VERSION = 1;
    public static final String DATABASE_NAME = "Wellcome.db";

    public WellcomeDbHelper(Context context){
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    public void onCreate(SQLiteDatabase db){
        db.execSQL(UserContract.SQL_CREATE_ENTRIES);
        db.execSQL(HostingContract.SQL_CREATE_ENTRIES);
    }

    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(UserContract.SQL_DELETE_ENTRIES);
        db.execSQL(HostingContract.SQL_DELETE_ENTRIES);
        onCreate(db);
    }

    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion){
        onUpgrade(db, oldVersion, newVersion);
    }
}

File: app/src/main/java/com/example/wellcome/utils/WellcomeDbHelper.kt
package com.example.wellcome.utils

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.example.wellcome.utils.Assistance.SQL_ASSISTANCE_CREATE_ENTRIES
import com.example.wellcome.utils.Assistance.SQL_ASSISTANCE_DELETE_ENTRIES
import com.example.wellcome.utils.Lesson.SQL_LESSON_CREATE_ENTRIES
import com.example.wellcome.utils.Lesson.SQL_LESSON_DELETE_ENTRIES
import com.example.wellcome.utils.Logement.SQL_LOGEMENT_CREATE_ENTRIES
import com.example.wellcome.utils.Logement.SQL_LOGEMENT_DELETE_ENTRIES

class WellcomeDbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION){
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(SQL_ASSISTANCE_CREATE_ENTRIES)
        db.execSQL(SQL_LESSON_CREATE_ENTRIES)
        db.execSQL(SQL_LOGEMENT_CREATE_ENTRIES)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(SQL_ASSISTANCE_DELETE_ENTRIES)
        db.execSQL(SQL_LESSON_DELETE_ENTRIES)
        db.execSQL(SQL_LOGEMENT_DELETE_ENTRIES)
        onCreate(db)
    }

    override fun onConfigure(db: SQLiteDatabase?) {
        db?.setForeignKeyConstraintsEnabled(true)
    }

    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }

    companion object {
        // If you change the database schema, you must increment the database version.
        const val DATABASE_VERSION = 3
        const val DATABASE_NAME = "Wellcome.db"
    }
}